### ISSUE TITLE
Add w3nco module to modulefiles

### ISSUE THREAD

**Issue by KateFriedman-NOAA** [NOAA-EMC/UPP#541]
While working on initial changes to introduce version files into the global-workflow `develop` branch I encountered an issue building the UPP (8767929) that checks out under the ufs-weather-model code (9ae617e). The following error occurs:

```
CMake Error at /apps/cmake/3.20.1/share/cmake-3.20/Modules/CMakeFindDependencyMacro.cmake:47 (find_package):
  Could not find a package configuration file provided by "w3nco" with any of
  the following names:

    w3ncoConfig.cmake
    w3nco-config.cmake

  Add the installation prefix of "w3nco" to CMAKE_PREFIX_PATH or set
  "w3nco_DIR" to a directory containing one of the above files.  If "w3nco"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  /scratch2/NCEPDEV/nwprod/hpc-stack/libs/hpc-stack/intel-2022.1.2/impi-2022.1.2/nemsio/2.5.2/lib/cmake/nemsio/nemsio-config.cmake:41 (find_dependency)
  CMakeLists.txt:56 (find_package)
```

Build log with error down below.

I hadn't made any changes to the ufs-weather-model or UPP. I thought it may have been because I changed the global-workflow `machine-setup.sh` to do a `module reset` instead of a `module purge` but when I change it back to a purge the same build error occurs.

I then added the `w3nco` module load to the UPP `hera.lua` file and the build succeeded with versions set from the workflow level. Here is the git diff:

```
diff --git a/modulefiles/hera.lua b/modulefiles/hera.lua
index a0e6574..5a4f0a5 100644
--- a/modulefiles/hera.lua
+++ b/modulefiles/hera.lua
@@ -41,6 +41,8 @@ crtm_ver=os.getenv("crtm_ver") or "2.3.0"
 load(pathJoin("crtm", crtm_ver))
 w3emc_ver=os.getenv("w3emc_ver") or "2.9.2"
 load(pathJoin("w3emc", w3emc_ver))
+w3nco_ver=os.getenv("w3nco_ver") or "2.4.1"
+load(pathJoin("w3nco", w3nco_ver))
 
 nemsio_ver=os.getenv("nemsio_ver") or "2.5.4"
 load(pathJoin("nemsio", nemsio_ver))
```

Here are my various build logs:
* UPP as-is (`module reset` and versions set in g-w level): /scratch1/NCEPDEV/global/Kate.Friedman/git/fork_feature-dev-wcoss2/sorc/logs/build_upp.log_test1
* Same as test1 but with `module purge` put back: /scratch1/NCEPDEV/global/Kate.Friedman/git/fork_feature-dev-wcoss2/sorc/logs/build_upp.log_test2
* Added `w3nco` to UPP modulefile (and used `module reset` and version files in g-w level): /scratch1/NCEPDEV/global/Kate.Friedman/git/fork_feature-dev-wcoss2/sorc/logs/build_upp.log

Pretty sure I'm setting the same module versions from g-w as the defaults in this version of the UPP, but please feel free to double check in the logs. This version of the UPP builds fine in the current global-workflow `develop` branch so it seems like the introduction of versions at the workflow level causes the need to add the `w3nco` module? Please let me know if you see a different error happening. Thanks!

**Comment by WenMeng-NOAA** [comment]
@KateFriedman-NOAA The UPP was updated with w3emc/2.9.2. There is no need for w3nco anymore.

**Comment by WenMeng-NOAA** [comment]
There is no issue to me to build UPP executable with build script under UPP/tests/compile_upp.sh on Hera. We might wait for @aerorahul 's comments.

**Comment by WenMeng-NOAA** [comment]
@KateFriedman-NOAA I recalled that I got the similar issue (see https://github.com/NOAA-EMC/UPP/pull/528). You might try use nemsio/2.5.4 instead of 2.5.2.

**Comment by WenMeng-NOAA** [comment]
@KateFriedman-NOAA 
I looked into your build log /scratch1/NCEPDEV/global/Kate.Friedman/git/fork_feature-dev-wcoss2/sorc/logs/build_upp.log_test1.
nemsio/2.5.2 was loaded. According to @aerorahul, nemsio/2.5.2 has a dependency on w3nco.  Your issue should be solved with loading nemsio/2.5.4. Please test it. Thanks!

**Comment by KateFriedman-NOAA** [comment]
@WenMeng-NOAA Using `nemsio/2.5.4` worked, thank you! I'll close this issue and readdress as needed when I get further into the work to add version files to global-workflow `develop`.
